official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -203,7 +211,19 @@ def step_fn(inputs):
             loss += (l2_loss / num_replicas)
           else:
             loss += (tf.reduce_sum(model.losses) / num_replicas)
+
+          # Scale the loss
+          if flags_obj.fp16_implementation == "graph_rewrite":
+            loss = loss * tf.cast(loss_scale, loss.dtype)
```

Comment:
Better to use [`opt.get_scaled_loss`](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/mixed_precision/experimental/LossScaleOptimizer#get_scaled_loss), and [`opt.get_unscaled_gradients`](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/mixed_precision/experimental/LossScaleOptimizer#get_unscaled_gradients) below.

Also, I would have the condition be `if flags_obj.dtype == "fp16"` instead, because this logic will also be needed for the tf.keras mixed precision API.